### PR TITLE
[Feature Req] Adding filter option in Join/Create Organisation

### LINKED ISSUE #741: Feature request for adding filter option in Create or Join Organisation Page
**Is your feature request related to a problem? Please describe.**
The feature allowed the user to filter organisation list whether they are public or private. This will help user to navigate easily between the organisation of their choice.

**Describe the solution you'd like**
A popup menu that filters the list based on user choice.

**Describe alternatives you've considered**
Tab menu to display public and private organisation separately.

**Approach to be followed (optional)**
The list will be fetched from backend based on user choice.

### PR DESCRIPTION
**Pull request by sumitra19jha**
**What kind of change does this PR introduce?**
Closes #741 feature request

**Did you add tests for your changes?**
Not required

**If relevant, did you update the documentation?**
Not required

**Summary**

**Does this PR introduce a breaking change?**
No

**Other information**

https://user-images.githubusercontent.com/35868598/116586940-9bf8c780-a937-11eb-90be-177e71b68915.mp4

### CODE REVIEW

**Comment by sumitra19jha** [comment]
@Sagar2366 This is the separate PR dealing with only filter feature

**Comment by Sagar2366** [comment]
@sumitra19jha thank you for the PR, please resolve conflicts.

**Review by dellyjm — COMMENTED** [review]
I'm not sure about the strings in the code.

File: lib/view_models/page_view_model/join_organization_view_model.dart [hunk 1]
```diff
@@ -174,24 +174,39 @@ class JoinOrgnizationViewModel extends BaseModel {
...
 
-  Future fetchOrg() async {
+  Future fetchOrg(String filter) async {
     getCurrentUserId();
     //function to fetch the org from the server
     final GraphQLClient _client = graphQLConfiguration.authClient();
-    final QueryResult organizationQueryResult = await _client
-        .query(QueryOptions(documentNode: gql(_query.fetchOrganizations)));
+    final QueryResult organizationQueryResult = await _client.query(
+      filter == 'Show All'
```
[the remark this answers is not in the document]
Comment (sumitra19jha) [reply lib/view_models/page_view_model/join_organization_view_model.dart]: @dellyjm The enum is added now and text strings are replaced

**Comment by sumitra19jha** [comment]
https://user-images.githubusercontent.com/35868598/117532209-1b575c80-b004-11eb-808c-c64547bab638.mp4

Enum is added now and the snapshot shows working with current modification @dellyjm

**Review by DeltaHarbinger — APPROVED** [review]

**Comment by DeltaHarbinger** [comment]
This PR has been merged. Thanks @sumitra19jha

**Comment by sumitra19jha** [comment]
Thank you @DeltaHarbinger

**Pull request merged**